src/AggregateFunctions/AggregateFunctionGroupSortedArray.h

```diff
@@ -0,0 +1,248 @@
+        size_t batch_size, AggregateDataPtr place, const IColumn ** columns, Arena * arena, ssize_t if_argument_pos) const override
+    {
+        State & data = this->data(place);
+        if constexpr (is_weighted)
+        {
+            StringRef ref = columns[1]->getRawData();
+            UInt64 values[batch_size];
+            memcpy(values, ref.data, batch_size * sizeof(UInt64));
+            UInt64 * bestRows = new UInt64[this->threshold];
+            size_t num_results;
+
+            //First store the first n elements with the column number
+            if (if_argument_pos >= 0)
+            {
+                UInt64 * value_w = values;
+
+                const auto & flags = assert_cast<const ColumnUInt8 &>(*columns[if_argument_pos]).getData();
+                for (size_t i = 0; i < batch_size; ++i)
+                {
+                    if (flags[i])
+                        *(value_w++) = values[i];
+                }
+
+                batch_size = value_w - values;
+            }
+
+            num_results = std::min(this->threshold, batch_size);
+            getFirstNElements(values, batch_size, num_results, bestRows);
+            for (size_t i = 0; i < num_results; i++)
+            {
+                auto row = bestRows[i];
+                data.add(readItem<T, is_plain_column>(columns[0], arena, row), values[row]);
+            }
+            delete[] bestRows;
+        }
+        else
+        {
+            if (if_argument_pos >= 0)
+            {
+                const auto & flags = assert_cast<const ColumnUInt8 &>(*columns[if_argument_pos]).getData();
+                for (size_t i = 0; i < batch_size; ++i)
+                    if (flags[i])
+                    {
+                        data.add(readItem<T, is_plain_column>(columns[0], arena, i));
+                    }
+            }
+            else
+            {
+                for (size_t i = 0; i < batch_size; ++i)
+                    data.add(readItem<T, is_plain_column>(columns[0], arena, i));
+            }
+        }
+    }
+
+    void merge(AggregateDataPtr __restrict place, ConstAggregateDataPtr rhs, Arena *) const override
+    {
+        this->data(place).merge(this->data(rhs));
+    }
+
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> /* version */) const override
+    {
+        this->data(place).serialize(buf);
+    }
+
+    void
+    deserialize(AggregateDataPtr __restrict place, ReadBuffer & buf, std::optional<size_t> /* version  */, Arena * arena) const override
+    {
+        this->data(place).deserialize(buf, arena);
+    }
+
+    void insertResultInto(AggregateDataPtr __restrict place, IColumn & to, Arena * /*arena*/) const override
+    {
+        ColumnArray & arr_to = assert_cast<ColumnArray &>(to);
+        ColumnArray::Offsets & offsets_to = arr_to.getOffsets();
+
+        auto & values = this->data(place).values;
+        int old_size = offsets_to.back();
```

Comment:
Please no `int`. `int` in general is a bad option for sizes since you don't know how big they are and if they are able to fit the size type. Either use the exact same size returned by back() or simply use `auto`.

Reply:
Yes, I don't need to declare 'old_size' as a variable anymore because we only use it once.